db/state/aggregator.go

```diff
@@ -1639,7 +1639,7 @@ func (a *Aggregator) BuildFilesInBackground(txNum uint64) chan struct{} {
 
 // Returns the first known txNum found in history files of a given domain
 func (at *AggregatorRoTx) HistoryStartFrom(name kv.Domain) uint64 {
-	return at.d[name].HistoryStartFrom()
+	return at.d[name].HistoryStartFrom(at.a.db)
```

Comment:
pass `tx` from caller - otherwise impossible to provide `consistenant data view` guaranties. 
txn must be created at highest-possible level - to make lower level funcs composable.

Reply:
This is a bit awkward because some `HistoryStartFrom` implementations have a `tx` as a receiver, e.g.  `temporal.tx` , so the extra parameter sometimes seems redundant.

Reply:
You want to have the interface method be `HistoryStartFrom(name kv.Domain, tx kv.Tx) uint64`, but this would means that in `kv_temporal.go` we would end up having:

```go
func (tx *Tx) HistoryStartFrom(name kv.Domain, roTx kv.Tx) uint64 {
	return tx.historyStartFrom(name, tx)
}
```

and 

```
func (tx *RwTx) HistoryStartFrom(name kv.Domain, roTx kv.Tx) uint64 {
	return tx.historyStartFrom(name, tx)
}
```

The `rotTx` would become a redundant parameter to the function because the function receiver is the transaction itself in these cases.

Reply:
Same thing applies to `DomainRoTx` .

Reply:
pls review again